src/ShellCheck/Analytics.hs

```diff
@@ -4903,5 +4904,27 @@ checkBatsTestDoesNotUseNegation params t =
             x:rest -> isLastOf t rest
             [] -> False
 
+prop_checkFunctionRedeclaration1 = verifyTree checkFunctionRedeclaration "f () { true; }; f () { true; }"
+prop_checkFunctionRedeclaration2 = verifyNotTree checkFunctionRedeclaration "f () { true; }; g () { true; }"
+
+checkFunctionRedeclaration params t =
+  map
```

Comment:
You could simplify this as `execWriter $ mapM (\ts -> warn (snd $ ts !! 1) 2216 "has already been defined twice") $ ..` instead of using `newTokenComment` from scratch